pkg/controller/attach_detach.go

```diff
@@ -800,6 +796,56 @@ func (r *ReconcileAttachDetach) recoverAzVolumeAttachment(ctx context.Context, r
 	return nil
 }
 
+func (r *ReconcileAttachDetach) isAttached(ctx context.Context, volumeHandle, nodeName string) (bool, error) {
+	w, _ := workflow.GetWorkflowFromContext(ctx)
+	disk, err := r.cloudDiskAttacher.CheckDiskExists(ctx, volumeHandle)
+	if disk == nil && err == nil {
+		// The rate limit of Get Disk call is 15000 times per 3 mins. We shouldn't get throttled often.
+		return false, fmt.Errorf("getting disk is throttled. %s", azureconstants.RateLimited)
+	}
+	if err != nil {
+		w.Logger().Errorf(err, "Volume (%s) not found", volumeHandle)
+		return false, err
+	}
+	// there is possibility that disk is nil when it is throttled
+	// don't check disk state when GetDisk is throttled
+	if disk != nil {
```

Comment:
When it's throttled, the function will return in line 804, so here `disk` won't be `nil`. We may keep the if statement for safe but update the comment to avoid confusion.